src/commands/configure/group.rb

```diff
@@ -49,7 +48,7 @@ def answer_file
         end
 
         def custom_configuration
-          cache.add(group_name)
+          GroupCache.update { |c| c.add group_name }
```

Comment:
Both here and in the tests we do similar things to this in several places, where we call `GroupCache.update` only to `add` a single node in the block. This seems a bit too verbose for what we're doing, and looks slightly confusing as it's not clear why we need the `update` if we're not doing anything else in the block passed to it.

A possible improvement would be to add an `add` class method to `GroupCache` to abstract away this pattern. This would then make it clearer I think that `GroupCache.add` will just immediately add a node to the group cache, whereas `GroupCache.transaction do ...` (new name as suggested below) will allow adding nodes to the group cache within a transaction which may or may not be saved.